server/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsServiceImpl.java

```diff
@@ -92,13 +155,13 @@
 
         Map<String, String> detailsMap;
 
-        final Answer answer = agentManager.easySend(hostId, command);
+        Answer answer = agentManager.easySend(hostId, command);
 
-        if (answer != null && (answer instanceof DiagnosticsAnswer)) {
+        if (answer != null) {
             detailsMap = ((DiagnosticsAnswer) answer).getExecutionDetails();
             return detailsMap;
         } else {
-            throw new CloudRuntimeException("Failed to execute diagnostics command on remote host: " + answer.getDetails());
+            throw new CloudRuntimeException("Failed to execute diagnostics command on remote host: " + vmInstance.getHostName());
```

Comment:
this can be a host, doesn't have to be a vmInstance, does it?

Reply:
The command is run on a systemvm (cpvm, ssvm, router), but the message logs the host name. I've reworded to make it somewhat better.

Reply:
ok, my bad